stdlib/ballerina-mime/src/main/java/org/ballerinalang/mime/util/MimeUtil.java

```diff
@@ -443,6 +460,44 @@ static boolean isNestedPartsAvailable(BStruct bodyPart) {
                 bodyPart.getNativeData(BODY_PARTS) != null;
     }
 
+    /**
+     * Extract overflow configuration settings from server annotations or get the default settings.
+     *
+     * @param context   Represent ballerina context
+     * @param configAnn Server configs
+     * @return A map with overflow settings
+     */
+    public static Map<String, Object> getOverflowSettings(Context context, Annotation configAnn) {
+        if (configAnn == null) {
+            return getDefaultOverflowSettings();
+        }
+        Map<String, Object> overflowSettings = new HashMap<>();
+        overflowSettings.put(ANN_CONFIG_MEMORY_THRESHOLD, getMemoryThresholdInBytes(configAnn));
+        String tempLocation = getTempLocation(configAnn);
+        if (tempLocation != null) {
+            overflowSettings.put(ANN_CONFIG_TEMP_LOCATION, tempLocation);
+        }
+        return overflowSettings;
+    }
+
+    private static String getTempLocation(Annotation configAnn) {
+        Struct overflowConfig = configAnn.getValue().getRefField(ANN_CONFIG_OVERFLOW_CONFIG).getStructValue();
+        return overflowConfig.getStringField(ANN_CONFIG_TEMP_LOCATION);
+    }
+
+    private static Long getMemoryThresholdInBytes(Annotation configAnn) {
+        Struct overflowConfig = configAnn.getValue().getRefField(ANN_CONFIG_OVERFLOW_CONFIG).getStructValue();
+        double memoryThreshold = overflowConfig.getFloatField(ANN_CONFIG_MEMORY_THRESHOLD);
+        memoryThreshold = memoryThreshold * (ONE_MB_IN_BYTES);
+        return Double.valueOf(memoryThreshold).longValue();
+    }
+
+    private static Map<String, Object> getDefaultOverflowSettings() {
+        Map<String, Object> overflowSettings = new HashMap<>();
+        overflowSettings.put(ANN_CONFIG_MEMORY_THRESHOLD, (long) BYTE_LIMIT);
+        return overflowSettings;
```

Comment:
Shall we use singletonMap method to make things efficient?
return Collections.singletonMap(ANN_CONFIG_MEMORY_THRESHOLD, (long) BYTE_LIMIT);